### PR TITLE
drivers: misc: a4988: implement driver for stepper controller

### PR DESCRIPTION
**Pull request by badochov**
Added driver support for stepper motor controller a4988.

### CODE REVIEW

File: dts/bindings/misc/allegro,a4988.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2021 Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
```
Comment (gmarull) [inline dts/bindings/misc/allegro,a4988.yaml:2]: hey what's going on here, I've never written this binding, or did I?

**Review by gmarull — CHANGES_REQUESTED** [review]
this probably deserves a proper driver class, all steppers will likely be driven the same way (or have a common ground)

File: dts/bindings/misc/allegro,a4988.yaml (hunk 1 above)
Comment (badochov) [reply dts/bindings/misc/allegro,a4988.yaml]: My bad forgot to adjust, fixed now.

File: drivers/misc/a4988/a4988.c [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+	return 0;
+}
+
+int a4988_reset(const struct device *dev, bool reset)
+{
+	const struct a4988_dev_config *config = dev->config;
+	int err;
+
+	err = gpio_pin_set_dt(&config->gpio.reset, !reset);
```
Comment (nixward) [inline drivers/misc/a4988/a4988.c:132]:
```suggestion
	err = gpio_pin_set_dt(&config->gpio.reset, reset);
```

There is a gpio flag to indicate a pin is used active low that can be used in an overlay or board dts file. Similar for sleep and enable pins. You should add the active low requirement to the binding description for each gpio pin.

**Pull request closed without merging**